hpx/runtime/actions/continuation.hpp

```diff
@@ -415,12 +392,18 @@ namespace hpx { namespace actions
           : continuation(std::move(gid)), f_(std::forward<F>(f))
         {}
 
-        template <typename F>
+        template <typename F,
+            typename Enable
+                = typename std::enable_if<
+                    !std::is_same<
+                        typename util::decay<F>::type, typed_continuation>::value
+                    >::type
+        >
```

Comment:
For some compilers, you'd need to have const, non-const and rvalue ref overloads here so that this constructor will not be picked as "best fit", that's why I decided to add this enable if here, we have similar constructs already in the code base (tuple, function).